operator/e2e/tests/cert_manager_test.go

```diff
@@ -0,0 +1,247 @@
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: grove-webhook-server-cert
+  namespace: grove-system
+spec:
+  secretName: grove-webhook-server-cert
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  issuerRef:
+    name: selfsigned-issuer
+    kind: ClusterIssuer
+  dnsNames:
+  - grove-operator
+  - grove-operator.grove-system
+  - grove-operator.grove-system.svc
+  - grove-operator.grove-system.svc.cluster.local
+`
+)
+
+func TestCertManagerIntegration(t *testing.T) {
+	// 1. Prepare cluster
+	ctx := context.Background()
+	clientset, restConfig, dynamicClient, cleanup := prepareTestCluster(ctx, t, 1)
+	defer cleanup()
+
+	// 2. Install cert-manager
+	deps, err := e2e.GetDependencies()
+	if err != nil {
+		t.Fatalf("Failed to get dependencies: %v", err)
+	}
+
+	cmConfig := &setup.HelmInstallConfig{
+		RestConfig:      restConfig,
+		ReleaseName:     deps.HelmCharts.CertManager.ReleaseName,
+		ChartRef:        deps.HelmCharts.CertManager.ChartRef,
+		ChartVersion:    deps.HelmCharts.CertManager.Version,
+		Namespace:       deps.HelmCharts.CertManager.Namespace,
+		CreateNamespace: true,
+		Wait:            true,
+		Timeout:         10 * time.Minute, // cert-manager needs time to download images and start
+		RepoURL:         deps.HelmCharts.CertManager.RepoURL,
+		Values: map[string]interface{}{
+			"installCRDs": true,
+		},
+		HelmLoggerFunc: logger.Debugf,
+		Logger:         logger,
+	}
+
+	logger.Info("Installing cert-manager...")
+	if _, err := setup.InstallHelmChart(cmConfig); err != nil {
+		t.Fatalf("Failed to install cert-manager: %v", err)
+	}
+
+	// Wait for cert-manager webhook to be ready (InstallHelmChart with Wait=true should handle it, but to be safe)
+	if err := utils.WaitForPodsInNamespace(ctx, cmConfig.Namespace, restConfig, 3, 2*time.Minute, 5*time.Second, logger); err != nil {
+		t.Fatalf("cert-manager pods not ready: %v", err)
+	}
+
+	// 3. Create Issuer and Certificate
+	logger.Info("Creating ClusterIssuer and Certificate...")
+	if err := applyYAMLContent(ctx, restConfig, certManagerIssuerYAML); err != nil {
+		t.Fatalf("Failed to apply ClusterIssuer: %v", err)
+	}
+
+	// Wait a bit for ClusterIssuer
+	time.Sleep(5 * time.Second)
+
+	if err := applyYAMLContent(ctx, restConfig, certManagerCertificateYAML); err != nil {
+		t.Fatalf("Failed to apply Certificate: %v", err)
+	}
+
+	// 4. Wait for Secret to be created
+	logger.Info("Waiting for certificate secret to be created...")
+	err = utils.PollForCondition(ctx, 1*time.Minute, 2*time.Second, func() (bool, error) {
+		_, err := clientset.CoreV1().Secrets("grove-system").Get(ctx, "grove-webhook-server-cert", metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("Certificate secret was not created: %v", err)
+	}
+
+	// 5. Update Grove to use the external secret via Helm upgrade
+	logger.Info("Upgrading Grove to use cert-manager secret...")
+
+	chartPath := "../../charts"
```

Comment:
Please follow the runtime.Caller(0) pattern used else where.  See CreateDefaultKaiQueues and waitForWebhookReady for examples.